packages/styles/src/mixins/_badge.scss

```diff
@@ -7,7 +7,10 @@
   border-color: transparent;
 
   @include utilities.high-contrast-mode() {
+    background-color: Highlight;
     border-color: Highlight;
+    color: HighlightText;
+    forced-color-adjust: none; // Disable "readability backplate" on blink browser that interferes with the colors on this case
```

Comment:
About this comment, this is the related thingy: https://github.com/MicrosoftEdge/MSEdgeExplainers/blob/main/Accessibility/HighContrast/explainer.md#ensuring-readability

Basically without forced-color-adjust, you have this intermediate box with a contrasted background-color. This is great in principle, but if you provide a color as I did (HighlightText for color and Highlight for background), the contrast will be inverted on blink-based browser, meanwhile it will work on other browsers not using this "backplate" technic. So instead of targetting a specific browser, we can disable this "backplate" with this property.